src/main/java/hudson/remoting/Launcher.java

```diff
@@ -283,12 +356,20 @@ public void setNoCertificateCheck(boolean ignored) throws NoSuchAlgorithmExcepti
     @Option(name="-version",usage="Shows the version of the remoting jar and then exits")
     public boolean showVersion = false;
 
+    /**
+     * The original calling convention takes two positional arguments: secret key and agent name.
+     * @deprecated use {@link #secret} and {@link #name}
+     */
+    @Argument
+    @Deprecated
+    public List<String> args = new ArrayList<>();
 
-    public static void main(String... args) throws Exception {
+    public static void main(String... args) throws IOException, InterruptedException {
         Launcher launcher = new Launcher();
         CmdLineParser parser = new CmdLineParser(launcher);
         try {
             parser.parseArgument(args);
+            normalizeArguments(launcher);
```

Comment:
Invariant: after calling `CmdlineParser#parseArguments`, call `normalizeArguments` to convert the old-positional arguments (if any) to the new-style named arguments. This is done here and one other place.